ads/improvedigital.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {writeScript, checkData, validateDataExists} from '../src/3p';
+
+const adImproveField = ['width', 'height', 'placement', 'optin', 'keyvalue'];
+const adImproveMandatoryField = ['placement'];
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function improvedigital(global, data)
+{
+  checkData(data, adImproveField);
+  validateDataExists(data, adImproveMandatoryField);
+
+  let url = 'https://ad.360yield.com' +
+      '/adj?' +
+      'p=' + encodeURIComponent(data.placement) +
+      '&w=' + encodeURIComponent(data.width) +
+      '&h=' + encodeURIComponent(data.height) +
+      '&optin=' + encodeURIComponent(data.optin) +
+      '&tz=' + new Date().getTimezoneOffset();
+
+  const value = data.keyvalue;
+  let newData = '';
+  const amps = '&';
+
+  if (value && value.length > 0) {
+    const keys = value.split('&');
+    for (let i = 0;i < keys.length;i++)
+    {
+      const segment = keys[i].split('=');
+      if (i > 0) {
+        newData += amps;
+      }
+      newData += segment[0] + '=' + encodeURIComponent(segment[1]);
```

Comment:
No, that's not the one I'm talking about here. Let's imaging the user has defined the following value for `data.keyvalue`: "abc&b=2". Based on your current code, you will have:

1. `keys[0] == 'abc'`
2. `split('=')` => `segment == ['abc']`
3. Which means, `segment[0] == 'abc'` and `segment[1] == undefined`
4. Finally resulting in `newData += 'abc=undefined'`

It didn't seem like what you actually want.

There are similar other issues as well. E.g. let's assume that the user-passed `data.keyvalue == '&a=1'`:

1. `keys[0] == ''`
2. `split('=')` => `segment=['']`
3. => `segment[0] = ''` and `segment[1] == undefined`
4. => `newData += '=undefined'`. Also not likely a desired thing.

This issue can be easily addressed by just doing `if (!keys[i]) {continue;}` in the for-loop.